src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/DateTimeValueResolver.php

```diff
@@ -74,6 +85,9 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
             }
             try {
                 $date = new $class($value);
+                if ($this->clock) {
+                    $date = $this->clock->withTimeZone($date->getTimezone())->now()->modify($value);
```

Comment:
The clock is UTC by default. We need to get the timezone by default or from the value for backward compatibility.